src/cast_core/src/Truncator.cpp

```diff
@@ -0,0 +1,67 @@
+// Copyright 2019-present MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cast_core/actors/Truncator.hpp>
+
+#include <memory>
+
+#include <yaml-cpp/yaml.h>
+
+#include <bsoncxx/json.hpp>
+
+#include <mongocxx/client.hpp>
+#include <mongocxx/collection.hpp>
+#include <mongocxx/database.hpp>
+
+#include <boost/log/trivial.hpp>
+#include <boost/throw_exception.hpp>
+
+#include <gennylib/Cast.hpp>
+#include <gennylib/MongoException.hpp>
+#include <gennylib/context.hpp>
+
+#include <value_generators/DocumentGenerator.hpp>
+
+
+namespace genny::actor {
+struct Truncator::PhaseConfig {
+    mongocxx::database database;
+    mongocxx::collection collection;
+    metrics::Operation truncateOperation;
```

Comment:
Are we collecting latency and throughput?  Do we get that by default?

Reply:
Does throughput even make sense in this context? We're rate limited, currently what we're collecting is the time it takes to delete the one document.

Reply:
I think throughput still makes sense, because this is part of a stress test that (when incorporated with all the other stresses), may not be able to keep up.

Reply:
@rtimmons, how does one track throughput using an operation?

Reply:
From the work done on the other genny workload I've implemented we do get throughput.